Add a context distinguisher to the MLS key exchange method (#116)
sframe-wg/sframe: draft-ietf-sframe-enc.md

To generate keys and nonces for SFrame, we use the MLS exporter function to
generate a `base_key` value for each MLS epoch.  Each member of the group is
assigned a set of KID values, so that each member has a unique `sframe_key` and
`sframe_salt` that it uses to encrypt with.  Senders may choose any KID value
within their assigned set of KID values, e.g., to accommodate multiple
uncoordinated outbound media streams at a sender.

~~~~~ pseudocode
base_key = MLS-Exporter("SFrame 1.0", "", AEAD.Nk)
~~~~~

For compactness, we do not send the whole epoch number.  Instead, we send only its
low-order `E` bits.  Note that `E` effectively defines a re-ordering window, since
no more than 2^E epochs can be active at a given time.  Receivers MUST be
prepared for the epoch counter to roll over, removing an old epoch when a new
epoch with the same E lower bits is introduced.  (Sender indices cannot be
similarly compressed.)

> Let `S` required to encode the size of the group, i.e., the smallest value such

Reviewer: Had to read this several times, still not sure I'm sure what the missing word is:
```suggestion
Let `S`, required to encode the size of the group, be the smallest value such
```
or perhaps
```suggestion
Let `S` be the bits required to encode the size of the group, i.e., the smallest value such
```

Reviewer: Not really convinced by either of my suggestions to be honest. Perhaps its been too long since I've read some proper math expressions?

> Let `S` be the number of bits requierd to encode a member index in the group,

Reviewer: ```suggestion
Let `S` be the number of bits required to encode a member index in the group,
```
